app/src/main/java/com/example/solidarity/fragments/EventsFragment.java

```diff
@@ -121,12 +167,53 @@ public void done(List<Event> events, ParseException e) {
                     return;
                 }
                 adapter.clear();
-                adapter.addAll(events);
+                for (Event ev: events) {
+                    getDist(latitude, longitude, ev.getLocation(), distances, ev);
+                }
                 swipeContainer.setRefreshing(false);
-
-
             }
         });
+    }
 
+    public void getDist(String fromLat, String fromLong, String toAddress,
+                        final ArrayList<Integer> distances, final Event event) {
+        String URL_4 = "https://maps.googleapis.com/maps/api/distancematrix/json?uni" +
+                "ts=imperial&origins=" + fromLat + "," + fromLong + "&destinations=" +
+                toAddress + "&key=" + API_KEY;
+        client.get(URL_4, new JsonHttpResponseHandler() {
+            @Override
+            public void onSuccess(int statusCode, Headers headers, JsonHttpResponseHandler.JSON json) {
+                JSONObject jsonObject = json.jsonObject;
```

Comment:
This function is not well done in terms of code quality. Follow the example you learned from your assignment and create a class and construct the class from this JSONObject